src/utils/objectUtils.ts

```diff
@@ -255,21 +271,18 @@ export function transformToKebabCase<Output, Input>(
     return object as TransformToKebabCaseReturnType<Input, Output>
   }
 
-  const transformKey = (key: string) =>
-    key
-      .replace(/([a-z])([A-Z])/g, '$1-$2') // transforms basic camelCase
-      .replace(/([A-Z])([A-Z][a-z])/g, '$1-$2') // transforms abbreviations
-      .replace(/_/g, '-') // transforms snake_case
-      .toLowerCase() // finally lowercase all
-
   return Object.entries(object as Record<string, unknown>).reduce(
-    (result, [key, value]) => ({
```

Comment:
original implementation violated "no destructuring of the accumulator" perf rule